File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        if (nums.empty()) return 0;
        unordered_map<int, bool> hashset;
        for(auto &x : nums)
            hashset[x] = true;
        int longest = 1;
        for(auto &x : nums) {
            if(hashset.find(x-1) == hashset.end()) {
                // start of a sequence
                int curseq = 1, cur = x;
                while (hashset.find(cur+1) != hashset.end()) {
                    curseq++;
                    cur++;
                }
                longest = max(curseq, longest);
            }
        }
        return longest;
    }
};

File: Data Structures & Algorithms/search-insert-position/submission-2.py
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        l = 0
        r = len(nums) - 1
        if target > nums[-1]: return len(nums)
        if target < nums[0]: return 0
        mid = (l + r) // 2
        while r >= l:
            mid = (l + r) // 2

            if nums[mid] < target:
                l = mid + 1
            elif nums[mid] > target:
                r = mid - 1
            else:
                return mid
        
        return l
